### PR TITLE
implement adapter.bulk_save

### PR DESCRIPTION
**Pull request by sravfeyn**
This is a subset of https://github.com/dimagi/commcare-hq/pull/20743/

I would like to get this in without waiting for that other PR, as it's needed for bulk pillow processing as well and is independent anyway.

### CODE REVIEW

File: corehq/apps/userreports/sql/adapter.py [hunk 1]
```diff
@@ -133,23 +133,39 @@ def get_distinct_values(self, column, limit):
...
         except Exception as e:
             self.handle_exception(doc, e)
 
-    def _save_rows(self, rows, doc):
-        rows = [
+    def bulk_save(self, docs):
+        rows = []
+        for doc in docs:
+            rows.extend(self.get_all_values(doc))
+        self.save_rows(rows)
```
Comment (snopoke) [inline corehq/apps/userreports/sql/adapter.py]: How does the error handling work here - or is that still to come?

**Comment by emord** [comment]
the other PR was merged. is this still needed?

File: corehq/apps/userreports/tests/utils.py [hunk 2]
```diff
@@ -146,3 +152,40 @@ def get_simple_xform():
...
+                elif isinstance(value, time):
+                    row[idx] = value.strftime("%H:%M:%S.%f").rstrip('0').rstrip('.')
+                elif isinstance(value, six.integer_types):
+                    row[idx] = str(value)
+                elif isinstance(value, (float, Decimal)):
+                    row[idx] = _convert_decimal_to_string(row[idx])
+                elif isinstance(value, six.string_types):
+                    row[idx] = value.encode('utf-8')
+                elif value is None:
+                    row[idx] = ''
```
[the remark this answers is not in the document]
Comment (sravfeyn) [reply corehq/apps/userreports/tests/utils.py]: I actually grabbed most of this code from existing icds test util code, I am planning to replace that with this. I will keep this in mind.

File: corehq/apps/userreports/tests/utils.py [hunk 3]
```diff
@@ -146,3 +152,40 @@ def get_simple_xform():
         'VT': 'VT',
     })
     return xform.tostring()
+
+
+def load_data_from_db(table_name):
```
[the remark this answers is not in the document]
Comment (sravfeyn) [reply corehq/apps/userreports/tests/utils.py]: Yeah, it's used in few other places as well.

File: corehq/apps/userreports/sql/adapter.py (hunk 1 above)
Comment (sravfeyn) [reply corehq/apps/userreports/sql/adapter.py]: Yes, planning to add one separately. There's usually a `save` and `best_effort_save` corresponding to these save methods that do error handling in this code for some reason.

**Comment by sravfeyn** [comment]
Closing since the other PR is merged now

**Pull request closed without merging**